LoRaEngine/LoraKeysManagerFacade/CreateEdgeDevice.cs

```diff
@@ -75,83 +71,17 @@ public CreateEdgeDevice(RegistryManager registryManager)
-                    {
-                        ["[$iot_hub_resource_id]"] = Environment.GetEnvironmentVariable("IOT_HUB_RESOURCE_ID"),
-                        ["[$log_analytics_workspace_id]"] = Environment.GetEnvironmentVariable("LOG_ANALYTICS_WORKSPACE_ID"),
-                        ["[$log_analytics_shared_key]"] = Environment.GetEnvironmentVariable("LOG_ANALYTICS_WORKSPACE_KEY")
-                    });
-
-                    _ = await this.registryManager.AddConfigurationAsync(new Configuration($"obs-{Guid.NewGuid()}")
-                    {
-                        Content = observabilityConfigurationContent,
-                        TargetCondition = $"deviceId='{deviceName}'"
-                    });
-                }
-
-                var twin = new Twin();
-                twin.Properties.Desired = new TwinCollection($"{{FacadeServerUrl:'https://{GetEnvironmentVariable("FACADE_HOST_NAME")}.azurewebsites.net/api/',FacadeAuthCode: '{facadeKey}'}}");
-                var remoteTwin = await this.registryManager.GetTwinAsync(deviceName);
-
-                _ = await this.registryManager.UpdateTwinAsync(deviceName, "LoRaWanNetworkSrvModule", twin, remoteTwin.ETag);
-
-                // This section will get deployed ONLY if the user selected the "deploy end device" options.
-                // Information in this if clause, is for demo purpose only and should not be used for productive workloads.
-                if (deployEndDevice)
-                {
-                    var otaaDevice = new Device(OtaaDeviceId);
-
-                    _ = await this.registryManager.AddDeviceAsync(otaaDevice);
-
-                    var otaaEndTwin = new Twin();
-                    otaaEndTwin.Properties.Desired = new TwinCollection(@"{AppEUI:'BE7A0000000014E2',AppKey:'8AFE71A145B253E49C3031AD068277A1',GatewayID:'',SensorDecoder:'DecoderValueSensor'}");
-                    var otaaRemoteTwin = _ = await this.registryManager.GetTwinAsync(OtaaDeviceId);
-                    _ = await this.registryManager.UpdateTwinAsync(OtaaDeviceId, otaaEndTwin, otaaRemoteTwin.ETag);
-
-                    var abpDevice = new Device(AbpDeviceId);
-                    _ = await this.registryManager.AddDeviceAsync(abpDevice);
-                    var abpTwin = new Twin();
-                    abpTwin.Properties.Desired = new TwinCollection(@"{AppSKey:'2B7E151628AED2A6ABF7158809CF4F3C',NwkSKey:'3B7E151628AED2A6ABF7158809CF4F3C',GatewayID:'',DevAddr:'0228B1B1',SensorDecoder:'DecoderValueSensor'}");
-                    var abpRemoteTwin = await this.registryManager.GetTwinAsync(AbpDeviceId);
-                    _ = await this.registryManager.UpdateTwinAsync(AbpDeviceId, abpTwin, abpRemoteTwin.ETag);
-                }
+                await this.registryManager.CreateEdgeDeviceAsync(
+                    deviceName,
+                    deployEndDevice,
+                    new Uri($"{siteUrl}/api/"),
```

Comment:
The better way to combine URLs is to use a [`Uri` constructor that accepts a base and relative part](https://docs.microsoft.com/en-us/dotnet/api/system.uri.-ctor#System_Uri__ctor_System_Uri_System_String_):

```suggestion
                    new Uri(siteUrl, "/api/"),
```

Reply:
in this case, the first / is not needed and should be removed.